File: frontend/src/types/review.ts
export interface ReviewTopic {
  id: string;
  name: string;
}

export interface Review {
  id: string;
  user_id: string;
  topic_id: string;
  topic: ReviewTopic;
  due_date: string;
  completed_at: string | null;
  status: 'PENDING' | 'COMPLETED';
  review_number: number;
  created_at: string;
}

export interface ReviewUpdate {
  status?: 'PENDING' | 'COMPLETED';
  completed_at?: string | null;
}

File: backend/app/models/review.py
import uuid
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base

class Review(Base):
    __tablename__ = "reviews"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    topic_id = Column(UUID(as_uuid=True), ForeignKey("topics.id", ondelete="CASCADE"), nullable=False)
    
    due_date = Column(DateTime(timezone=True), nullable=False)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    status = Column(String, default="PENDING", nullable=False) # PENDING, COMPLETED
    review_number = Column(Integer, nullable=False) # 1, 2, 3...
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", back_populates="reviews")
    topic = relationship("Topic", back_populates="reviews")
